ReSharper.FSharp/src/FSharp.Psi.Features/src/Parsing/FSharpImplTreeBuilder.fs

```diff
@@ -688,15 +654,42 @@ type FSharpImplTreeBuilder(lexer, document, decls, lifetime, path, projectedOffs
 
         x.Done(binding.RangeOfBindingWithRhs, mark, ElementType.TOP_BINDING)
 
-    member x.ProcessTopLevelBindings(bindings) =
+    member x.ProcessTopLevelBindings(bindings, range) =
         match bindings with
         | [] -> ()
+        | [SynBinding(kind = SynBindingKind.Do; expr = expr)] ->
+            let mark = x.Mark(range)
+            x.AdvanceToTokenOrRangeStart(FSharpTokenType.DO, range)
+            let expr = x.RemoveDoExpr(expr)
+            x.MarkChameleonExpression(expr)
+            x.Done(range, mark, ElementType.DO_STATEMENT)
+
+        // `extern` declarations are represented as normal `let` bindings with fake rhs expressions in FCS AST.
+        // This is a workaround to mark such declarations and not to mark the non-existent expressions inside it.
+        | [SynBinding(attributes = attrs; headPat = headPat; returnInfo = returnInfo; trivia = trivia; xmlDoc = XmlDoc xmlDoc)] when
+            trivia.LetKeyword.IsNone ->
```

Comment:
Just a heads up, but in more recent versions of the compiler [extern](https://fsharp.github.io/fsharp-compiler-docs/reference/fsharp-compiler-syntaxtrivia-synleadingkeyword.html#Extern) is captured as an actual DU.

Reply:
Thanks! We'll use it once we update the compiler.